File: GameCore/src/game/core/ISizeable.java
package game.core;

/**
 * Доска игры может быть разных рамеров.
 *
 * @author <a href="mailto:vladimir.romanov@gmail.com">Romanov V.Y.</a>
 */
public interface ISizeable {
    /**
     * Выдать допустимые для данной игры размеры доски.
     *
     * @return массив пар (ширина, высота) доски.
     */
    int[][] getSizes();
}

File: GameCore/src/game/core/ISizeable.kt
package game.core

/**
 * Доска игры может быть разных рамеров.
 *
 * @author [Romanov V.Y.](mailto:vladimir.romanov@gmail.com)
 */
interface ISizeable {
    /**
     * Выдать допустимые для данной игры размеры доски.
     *
     * @return массив пар (ширина, высота) доски.
     */
    val sizes: Array<IntArray>
}